tests/e2e/common/diagnosticcollector.go

```diff
@@ -0,0 +1,465 @@
+# limitations under the License.
+**/
+
+package common
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+	nfdclient "sigs.k8s.io/node-feature-discovery/pkg/generated/clientset/versioned"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	// the core group
+	Pods       = "pods"
+	Nodes      = "nodes"
+	Namespaces = "namespaces"
+
+	// the apps group
+	Deployments = "deployments"
+	DaemonSets  = "daemonsets"
+
+	// the batch group
+	Jobs = "jobs"
+
+	// Supported extensions
+	NodeFeature     = "nodeFeature"
+	NodeFeatureRule = "nodeFeatureRule"
+)
+
+type Collector interface {
+	Collect(context.Context) error
+}
+
+type nodes struct {
+	*Config
+}
+
+type namespaces struct {
+	*Config
+}
+
+type pods struct {
+	*Config
+}
+
+type deployments struct {
+	*Config
+}
+
+type daemonsets struct {
+	*Config
+}
+
+type jobs struct {
+	*Config
+}
+
+type nodeFeatures struct {
+	*Config
+	NfdClient *nfdclient.Clientset
+}
+
+type nodeFeatureRules struct {
+	*Config
+	NfdClient *nfdclient.Clientset
+}
+
+type Config struct {
+	clientset *kubernetes.Clientset
+	NfdClient *nfdclient.Clientset
+
+	artifactDir string
+	namespace   string
+
+	log    io.Writer
+	errLog io.Writer
+}
+
+func (c *Config) createFile(fp string) (io.WriteCloser, error) {
+	outfile, err := os.Create(filepath.Join(c.artifactDir, c.namespace, fp))
+	if err != nil {
+		return nil, fmt.Errorf("error creating %v: %w", fp, err)
+	}
+	return outfile, nil
+}
+
+func (c *Config) writeToFile(w io.Writer, data interface{}) error {
+	// Marshal data to YAML format
+	yamlBytes, err := yaml.Marshal(data)
+	if err != nil {
+		if _, wErr := fmt.Fprintf(c.errLog, "Error marshalling data: %v\n", err); wErr != nil {
+			return errors.Join(err, wErr)
+		}
+	}
+
+	// Write marshaled bytes to the provided io.Writer
+	_, err = w.Write(yamlBytes)
+	if err != nil {
+		if _, wErr := fmt.Fprintf(c.errLog, "Error writing data: %v\n", err); wErr != nil {
+			return errors.Join(err, wErr)
+		}
+	}
+
+	return nil
+}
+
+func (c nodes) Collect(ctx context.Context) error {
+	nodes, err := c.clientset.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		if _, wErr := fmt.Fprintf(c.errLog, "Error getting nodes: %v\n", err); wErr != nil {
+			return errors.Join(err, wErr)
+		}
+	}
+
+	nodesFile, err := c.createFile("nodes.yaml")
+	if err != nil {
+		return err
+	}
+	defer nodesFile.Close()
+
+	if err = c.writeToFile(nodesFile, nodes); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c namespaces) Collect(ctx context.Context) error {
+	namespaces, err := c.Config.clientset.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		if _, wErr := fmt.Fprintf(c.errLog, "Error getting namespaces: %v\n", err); wErr != nil {
+			return errors.Join(err, wErr)
+		}
+	}
```

Comment:
Assuming that we can configure `klog` to use `errLog` we could just do:
```suggestion
	if err != nil {
		klog.ErrorS(err, "Error getting namepaces")
	}
```